pkg/util/sets/set.go

```diff
@@ -117,6 +117,26 @@ func (s Set[T]) Difference(s2 Set[T]) Set[T] {
 	return result
 }
 
+// DifferenceIntersection returns the Difference and Intersection.
+// Usage scenario: When calculating the Difference, also want the deleted items(Intersection) from s.
+// For example:
+// s  = {a1, a2, a3}
+// s2 = {a1, a2, a4, a5}
+// s.DifferenceIntersection(s2) = {a3}, {a1, a2}
+// s2.DifferenceIntersection(s) = {a4, a5}, {a1, a2}
+func (s Set[T]) DifferenceIntersection(s2 Set[T]) (Set[T], Set[T]) {
```

Comment:
@hzxuzhonghu Does this make sense? Do we keep this function? I think it makes sense, although not currently used.